src/misc/cross.cpp

```diff
@@ -715,6 +715,88 @@ std::string cfstr_to_string(CFStringRef source)
 	return lang;
 }
 
+#if defined (__linux__)
+
```

Comment:
Suggest using these defines:

```c++
// Defined if compiling for OS from BSD family
#mesondefine BSD

// Defined if compiling for Linux (non-Android)
#mesondefine LINUX

// Defined if compiling for macOS
#mesondefine MACOSX

// Defined if compiling for Windows (any option)
#mesondefine WIN32
```

Looks like they're used in other code, like:

```
src/gui/sdlmain.cpp:#if defined(BSD) || defined(LINUX)
```

and

```
src/hardware/serialport/libserial.cpp:#if defined (LINUX) || defined (MACOSX) || defined (BSD)
```

(and others)

Reply:
We're moving away from Meson to CMake, so I don't think that's a good idea.

Reply:
:man_shrugging: it's just consistent with the existing code base.  If we're planning to rename these with CMake indentifiers (whatever those might be), then the conversion still is done consistently with a mass search-and-replace.

Reply:
Nah, you're right @interloper98. I haven't actually checked the sources. Yeah, let's go ahead with this, then we mass replace it when (if) needed.

Reply:
Sounds good.  I was already using the correct ones for Mac and Windows.  Changed the Linux macro as you noted.